vendor/github.com/docker/distribution/registry/storage/linkedblobstore.go

```diff
@@ -289,11 +281,19 @@ func (lbs *linkedBlobStore) Enumerate(ctx context.Context, ingestor func(digest.
 	return nil
 }
 
-func (lbs *linkedBlobStore) mount(ctx context.Context, sourceRepo reference.Named, dgst digest.Digest) (distribution.Descriptor, error) {
+func (lbs *linkedBlobStore) mount(ctx context.Context, repoOverride distribution.RepositoryOverride, sourceRepo reference.Named, dgst digest.Digest) (distribution.Descriptor, error) {
 	repo, err := lbs.registry.Repository(ctx, sourceRepo)
 	if err != nil {
 		return distribution.Descriptor{}, err
 	}
+
+	if repoOverride != nil {
+		repo, err = repoOverride(ctx, repo)
```

Comment:
> was the intent to stat the source one first and fall back to our wrapped repo in failure cases?

the intent was just to use the wrapped one which, internally, first stats the source local repo and then falls back to pullthrough stat for corresponding source image stream